lib/elm-test.js

```diff
@@ -92,26 +91,23 @@ function resolveFilePath(filename) {
 }
 
 let pathToElmBinary;
-let elmVersion = null;
 
 if (args.compiler === undefined) {
   try {
     pathToElmBinary = which.sync('elm');
   } catch (error) {
-    throw new Error(
+    console.error(
       `Cannot find elm executable, make sure it is installed.
 (If elm is not on your path or is called something different the --compiler flag might help.)`
     );
+    process.exit(1);
   }
 } else {
   try {
     pathToElmBinary = path.resolve(which.sync(args.compiler));
-
-    elmVersion = spawn
-      .sync(pathToElmBinary, ['--version'], { encoding: 'utf-8' })
-      .stdout.trim();
   } catch (error) {
-    throw new Error('The elm executable passed to --compiler must exist.');
+    console.error('The elm executable passed to --compiler must exist.');
+    process.exit(1);
```

Comment:
These errors were uncaught, which resulted in ugly console output (with full stack trace and all). Now they’re handled the same way as when we validate `args.reporter` somewhere else in this file.

Reply:
Thanks!

Reply:
Merged in https://github.com/rtfeldman/node-test-runner/pull/448